src/api/alchemy-external-wallet.ts

```diff
@@ -0,0 +1,166 @@
+
+  /**
+   * Returns the balance of this wallet at blockTag.
+   *
+   * @param blockTag The block to check the balance of
+   * @override
+   */
+  getBalance(blockTag?: BlockTag): Promise<BigNumber> {
+    return this.signer.getBalance(blockTag);
+  }
+
+  /**
+   * Returns the number of transactions this account has ever sent. This is the
+   * value required to be included in transactions as the nonce.
+   *
+   * @param blockTag The block to check the transaction count on
+   * @override
+   */
+  getTransactionCount(blockTag?: BlockTag): Promise<number> {
+    return this.signer.getTransactionCount(blockTag);
+  }
+
+  /**
+   * Returns the result of estimating the cost to send the transactionRequest,
+   * with this account address being used as the from field.
+   *
+   * @param transaction Transaction to estimate the gas on
+   * @override
+   */
+  estimateGas(transaction: Deferrable<TransactionRequest>): Promise<BigNumber> {
+    return this.signer.estimateGas(transaction);
+  }
+
+  /**
+   * Returns the result of calling using the transactionRequest, with this
+   * account address being used as the from field.
+   *
+   * @param transaction To make a call on
+   * @param blockTag The block to make the call on
+   * @override
+   */
+  call(
+    transaction: Deferrable<TransactionRequest>,
+    blockTag?: BlockTag
+  ): Promise<string> {
+    return this.signer.call(transaction, blockTag);
+  }
+
+  /**
+   * Populates all fields in a transaction, signs it and sends it to the network
+   *
+   * @override
+   */
+  sendTransaction(
+    transaction: Deferrable<TransactionRequest>
+  ): Promise<TransactionResponse> {
+    return this.signer.sendTransaction(transaction);
+  }
+
+  /**
+   * Returns the chain ID this wallet is connected to.
+   *
+   * @override
+   */
+  getChainId(): Promise<number> {
+    return this.signer.getChainId();
+  }
+
+  /**
+   * Returns the current gas price.
+   *
+   * @override
+   */
+  getGasPrice(): Promise<BigNumber> {
+    return this.signer.getGasPrice();
+  }
+
+  /**
+   * Returns the current recommended FeeData to use in a transaction.
+   *
+   * For an EIP-1559 transaction, the maxFeePerGas and maxPriorityFeePerGas
+   * should be used.
+   *
+   * For legacy transactions and networks which do not support EIP-1559, the
+   * gasPrice should be used.
+   *
+   * @override
+   */
+  getFeeData(): Promise<FeeData> {
+    return this.signer.getFeeData();
+  }
+
+  /**
+   * Looks up the address of name. If the name is not owned, or does not have a
+   * Resolver configured, or the Resolver does not have an address configured,
+   * null is returned.
+   *
+   * @param name Name of the ENS address
+   * @override
+   */
+  resolveName(name: string): Promise<string> {
+    return this.signer.resolveName(name);
+  }
+
+  /** Forwards any relevant request to the send method in the Provider */
+  send(method: string, params: any[]): Promise<any> {
+    return this.provider.send(method, params);
```

Comment:
Did you want this to be
```ts
return this.alchemy.provider.send(method, params)
```
or something like that?

Reply:
It actually needs to send the requests to the provider of the wallet, in order to properly handle queries like eth_requestAccounts! 

We could build in support for sending requests to the Alchemy Provider by having a method called sendAlchemy() or something...but in my head, if they really want the Alchemy Provider, they should just create a separate Alchemy object

Reply:
Update: I updated the comments on both the ExternalWallet and send() to accurately reflect that it will just send requests to the ExternalProvider. We won't support the Alchemy object in this package at this time (though later on we might)